Enforce the native pipe by default (#2796)
r-lib/lintr: NEWS.md

   + `linter=` argument of `Lint()`.
   + `with_defaults()`.
   + Linters `closed_curly_linter()`, `open_curly_linter()`, `paren_brace_linter()`, and `semicolon_terminator_linter()`.

> * The default for `pipe_consistency_linter()` is changed from `"auto"` (just be consistent) to `"|>"` (R native pipe required) to coincide with the same change in the Tidyverse Style Guide (#2707, @MichaelChirico).

Reviewer: ```suggestion
* The default for `pipe_consistency_linter()` is changed from `"auto"` (consistent choice of pipe required) to `"|>"` (R native pipe required) to coincide with the same change in the Tidyverse Style Guide (#2707, @MichaelChirico).
```
